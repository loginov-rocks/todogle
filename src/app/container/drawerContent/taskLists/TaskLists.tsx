import { ListItem, ListItemText } from '@material-ui/core';
import * as React from 'react';
import { useSelector } from 'react-redux';
import { useHistory } from 'react-router-dom';

import * as R from '../../../../routes';
import { getTaskListsArray } from '../../../../store/selectors';

interface Props {
  onTaskListClick?: () => void;
}

const TaskLists = ({ onTaskListClick }: Props) => {
  const history = useHistory();
  const taskLists = useSelector(getTaskListsArray);

  return (
    <>
      {taskLists.map(taskList => (
        <ListItem
          button
          key={taskList.id}
          onClick={() => {
            history.push(R.toTaskList(taskList.id));
            if (onTaskListClick) {
              onTaskListClick();
            }
          }}
        >
          <ListItemText primary={taskList.title} />
        </ListItem>
      ))}
    </>
  );
};

export default TaskLists;
