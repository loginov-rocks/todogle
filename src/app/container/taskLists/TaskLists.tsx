import {
  CircularProgress, List, ListItem, ListItemText,
} from '@material-ui/core';
import * as React from 'react';

import { TaskListResource } from '../../../services/gapi/TaskListResource';

interface Props {
  areLoaded: boolean;
  onClick: (taskListId: string) => void;
  taskLists: TaskListResource[];
}

const TaskLists = ({ areLoaded, onClick, taskLists }: Props) => {
  if (!areLoaded) {
    return <CircularProgress />;
  }

  return (
    <List>
      {taskLists.map(taskList => (
        <ListItem
          button
          key={taskList.id}
          onClick={() => onClick(taskList.id)}
        >
          <ListItemText primary={taskList.title} />
        </ListItem>
      ))}
    </List>
  );
};

export default TaskLists;
