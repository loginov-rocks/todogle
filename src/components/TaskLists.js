import React from 'react';

import gapi from '../services/gapi';
import SignOutButton from './SignOutButton';

export default class TaskLists extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isFetched: false,
      taskLists: null,
    };
  }

  componentDidMount() {
    gapi.getTaskLists()
      .then((taskLists) => {
        this.setState({
          isFetched: true,
          taskLists,
        });
      });
  }

  render() {
    const { isFetched, tasksLists } = this.state;

    return (
      <>
        <SignOutButton />
        {isFetched && (
          <pre>
            {JSON.stringify(tasksLists, null, 2)}
          </pre>
        )}
      </>
    );
  }
}
