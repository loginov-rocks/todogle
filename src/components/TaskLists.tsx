import * as React from 'react';

import gapi from '../services/gapi';
import TaskList from '../services/gapi/TaskList';
import SignOutButton from './SignOutButton';

interface State {
  isFetched: boolean,
  taskLists: TaskList[],
}

export default class TaskLists extends React.Component<{}, State> {
  constructor(props: {}) {
    super(props);

    this.state = {
      isFetched: false,
      taskLists: [],
    };
  }

  componentDidMount() {
    gapi.getTaskLists()
      .then(taskLists => {
        this.setState({
          isFetched: true,
          taskLists,
        });
      });
  }

  render() {
    const { isFetched, taskLists } = this.state;

    return (
      <>
        <SignOutButton />
        {isFetched && (
          <pre>
            {JSON.stringify(taskLists, null, 2)}
          </pre>
        )}
      </>
    );
  }
}
