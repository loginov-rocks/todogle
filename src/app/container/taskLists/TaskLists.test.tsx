import { shallow } from 'enzyme';
import * as React from 'react';

import taskLists from '../../../services/gapi/__fixtures__/taskLists.json';

import TaskLists from './TaskLists';

it('matches snapshot', () => {
  const wrapper = shallow(
    <TaskLists areLoaded onClick={() => undefined} taskLists={taskLists} />,
  );

  expect(wrapper).toMatchSnapshot();
});
