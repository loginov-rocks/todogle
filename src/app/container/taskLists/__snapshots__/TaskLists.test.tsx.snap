// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<WithStyles(ForwardRef(List))>
  <WithStyles(ForwardRef(ListItem))
    button={true}
    key="RUVUM3JvcDUxTUNHcDUtdw"
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(ListItemText))
      primary="The best task list"
    />
  </WithStyles(ForwardRef(ListItem))>
</WithStyles(ForwardRef(List))>
`;
